pandas/core/arrays/datetimelike.py

```diff
@@ -775,8 +775,10 @@ def searchsorted(self, value, side="left", sorter=None):
         if isinstance(value, str):
             try:
                 value = self._scalar_from_string(value)
-            except ValueError:
-                raise TypeError("searchsorted requires compatible dtype or scalar")
+            except ValueError as e:
+                raise TypeError(
+                    "searchsorted requires compatible dtype or scalar"
+                ) from e
```

Comment:
nitpick: can you make this "err" instead of "e"; i really dont like 1-letter names

Reply:
Eh...I would leave it alone for this PR.  I understand having an aesthetic preference, but `e` is strewn all over the place in this file, and I don't see why we should nitpick this one instance and hold this PR up because of that.

If we have a generally accepted (for this repository) a stylistic preference for variable names for exceptions, then fine, but I'm not aware of that at this point.

Reply:
yah, not a deal-breaker.  i think of this as a "if you have to push another commit anyway, this would be nice"